generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/FieldSpec.java

```diff
@@ -41,43 +45,37 @@ private FieldSpec(
         DateTimeRestrictions dateTimeRestrictions,
         FormatRestrictions formatRestrictions,
         MustContainRestriction mustContainRestriction,
-        FieldSpecSource source) {
+        @NotNull FieldSpecSource source) {
+
         this.setRestrictions = setRestrictions;
-        this.numericRestrictions = numericRestrictions;
-        this.stringRestrictions = stringRestrictions;
         this.nullRestrictions = nullRestrictions;
-        this.typeRestrictions = typeRestrictions;
-        this.dateTimeRestrictions = dateTimeRestrictions;
-        this.formatRestrictions = formatRestrictions;
         this.mustContainRestriction = mustContainRestriction;
-        this.source = source;
-    }
-
-    public SetRestrictions getSetRestrictions() {
-        return setRestrictions;
-    }
-
-    public NumericRestrictions getNumericRestrictions() {
-        return numericRestrictions;
-    }
-
-    public StringRestrictions getStringRestrictions() {
-        return stringRestrictions;
-    }
+        this.formatRestrictions = formatRestrictions;
 
-    public NullRestrictions getNullRestrictions() {
-        return nullRestrictions;
-    }
+        if (setRestrictions != null && setRestrictions.getWhitelist() != null && setRestrictions.getWhitelist().size() > 0) {
+            this.numericRestrictions = null;
```

Comment:
so this assumes that the date (etc) restrictions have already been applied to the set restriction.

I think it would be safer to do this trimming of other restrictions at that point instead of now, so there's no points where it can get out of sync

Reply:
e.g. merging field specs [set [1, 2, 3]] with [not numeric]
if this is done by creating the merged fieldspec then checking for contradictions. this constructor could remove the not numeric, before realising that it needed to remove everything from the set.

one alternative, would be making this constructor throw an exception if it gets both a setrestriction, and other restrictions.
and instead do this checking during merge elsewhere.

though this could be too large a change for now